From facd6a82d0ba6f0138fcca5f5e4cdd385401dd44 Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@consol.de>
Date: Tue, 20 Feb 2024 23:17:39 +0100
Subject: [PATCH] fix backslashes in plugin_output

Previously naemon required backslashes to be unescaped in order to appear correctly.
Seems like this is no longer required and backslashes can be passed through like they
are.

- introduced by #99
---
 Changes                           | 3 +++
 neb_module_naemon/result_thread.c | 7 +------
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/Changes b/Changes
index f0afbdb..def1d6a 100644
--- a/Changes
+++ b/Changes
@@ -1,5 +1,8 @@
 This file documents the revision history for mod_gearman.
 
+next:
+          - fix backslashes in plugin_output
+
 5.1.3 Tue Oct 17 23:32:07 CEST 2023
           - check_dummy: fix crash if output is empty
           - check_dummy: support putting return code in quotes
diff --git a/neb_module_naemon/result_thread.c b/neb_module_naemon/result_thread.c
index 58bb15b..3ee47f6 100644
--- a/neb_module_naemon/result_thread.c
+++ b/neb_module_naemon/result_thread.c
@@ -215,15 +215,10 @@ void *get_results( gearman_job_st *job, __attribute__((__unused__)) void *contex
                 char *tmp_newline = replace_str(value, "\\n", "\n");
                 if (tmp_newline == NULL)
                     chk_result->output = gm_strdup("(null)");
-
-                char *tmp_backslash = replace_str(tmp_newline, "\\\\", "\\");
-                if (tmp_backslash == NULL)
-                    chk_result->output = gm_strdup("(null)");
                 else
-                    chk_result->output = gm_strdup( tmp_backslash );
+                    chk_result->output = gm_strdup( tmp_newline );
 
                 gm_free(tmp_newline);
-                gm_free(tmp_backslash);
             }
         }
 
